core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -412,7 +412,6 @@ object CachedDataFrame
     val buffer = new Array[Byte](4 << 10) // 4K
     // final output is written to this buffer
     val output = new ByteBufferDataOutput(4 << 10,
-      DirectBufferAllocator.instance(),
```

Comment:
We should document this fact and to set JVM property -XX:MaxDirectMemorySize=<1024MB> as best practice.  1024 from default spark.driver.maxResultSize